### PR TITLE
agat: split from apple2.cpp, implement basic agat7 hardware.

### PR DESCRIPTION
**Pull request by shattered**

### CODE REVIEW

**Comment by rb6502** [comment]
I like this in general, but I have some comments.

1) Just create another subdevice for the agat7 that loads the proper ROM; making it a BIOS selection on the Disk II is completely wrong.
2) Unless the video is truly 100% on a card like PCs, the video should be in mame/video rather than under a2bus.  (And if it *is* on a card, load the chargen and other necessary ROMs as part of the card device instead of the driver).

File: src/mame/drivers/agat.cpp [hunk 1]
```diff
@@ -0,0 +1,1107 @@
...
+	- native keyboards (at least two variants)
+	- 840K floppy controller (MFM encoding, but track layout is unique)
+	- agat7: 64K and 128K onboard memory configurations
+	- agat9
+	- 3rd party slot devices
+
+************************************************************************/
+
+#include "emu.h"
+#include "includes/apple2.h"
```
Comment (ajrhacker) [inline src/mame/drivers/agat.cpp]: It doesn't seem clear why this dependency should be necessary when the driver has been split and agat_state doesn't inherit from the apple2 driver class.

Comment (shattered) [inline src/mame/drivers/agat.cpp]: AY3600.  I've not written native keyboard yet.

**Comment by shattered** [comment]
Video is onboard, but CPU is on a card :)

Submitted new PR, https://github.com/mamedev/mame/pull/2153

**Pull request closed without merging**